### PR TITLE
Updating a bunch of stuff

### PR DESCRIPTION
**Pull request by RaphaelIT7**
[+] Added `FCVAR_LUA_CLIENT` and `FCVAR_LUA_SERVER` flags
[+] Added missing Gmod classes & Interfaces
- `IGMHTML`
- `IGarrysMod`
- `IGet`
- `IGModAudioChannel`
- `IBassAudioStream`
- `IGMod_Audio`
- `IMenuSystem`
- `IResources`
- `IServerAddons`
- `IIntroScreen`
- `CGMOD_Player`
- `CGMODRules`
- `IGMODDataTable`
- `CVideoHolly` & `CVideoWriter`
- `IEngineAPI` (Gmod has `SetHeadtrackDLLName`)
- `CScriptedEntity`
- `CSENT_anim`
- `CSENT_brush`
- `CSENT_filter`
- `CSENT_point`
- `CWeaponSWEP`
- `CLuaCurlHTTPRequest`
- `CLuaUser`
- `GModDataPack`
- `CGModScreenspaceEffects`
- `CHudGMod`
- `ClientModeGModNormal`

[#] Updated Map limits
[#] Updated `MAX_PLAYERS` to 128
[#] Updated classes & Interfaces
- `CBaseEntity`
- `CServerGameClients`
- `CServerGameDLL`
- `IServerGameClients`
- `IServerGameDLL`
- `CAmmoDef`
- `Addon::FileSystem`
- `GameDepot::System`
- `Gamemode::System`
- `CLanguage`
- `IVEngineClient013`
- `IVEngineServer`
- `IShadowMgr`
- `IVModelRender`
- `IMaterial`
- `IMatRenderContext`
- `IShaderAPI`
- `IShaderDynamicAPI`
- `CVProfile`
- `CVProfCounter`
- `ConVar`
- `ISurface`
- `ICommandLine`
- `IMatSystemSurface`
- `CBaseCombatWeapon`
- `C_BaseEntity`
- `IMaterialProxy`
- `IBaseClientDLL`

### CODE REVIEW

**Comment by RaphaelIT7** [comment]
Everything should be fine but if something is missing/broken I should be able to fix it quickly.

Small side note: 
It think CBasePlayer or CBaseCombatCharacter is broken, but I could be wrong
Maybe in the next few days I find enouth time to look into it.

File: public/GarrysMod/GModDataPack.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+abstract_class GModDataPack
+{
+public:
+	virtual void* GetFromDatatable( std::string const& ) = 0;
+	virtual void* GetHashFromDatatable( std::string const& ) = 0;
+	virtual void* GetHashFromString( char const*, unsigned long ) = 0;
+	virtual void FindInDatatable( std::string const&, std::vector<LuaFindResult>&, bool ) = 0;
+	virtual void* FindFileInDatatable( std::string const&, bool, bool ) = 0;
+	virtual bool IsSingleplayer() = 0;
+	virtual void v000000000000000000000000000oo0000000000000000fff000000000000000000000000000000000o0o() = 0;
```
Comment (danielga) [inline public/GarrysMod/GModDataPack.h]: Can you get a better name for this one? If you don't know its name, I'd rather it be named something like `UnknownMethod`.

**Review by danielga — CHANGES_REQUESTED** [review]

File: public/GarrysMod/GModDataPack.h (hunk 1 above)
Comment (RaphaelIT7) [reply public/GarrysMod/GModDataPack.h]: It's how it's named in Gmod but I can change it
image

Comment (RaphaelIT7) [reply public/GarrysMod/GModDataPack.h]: I changed it to UnknownMethod but also left a comment with the original name

**Review by danielga — APPROVED** [review]

**Comment by RaphaelIT7** [comment]
Should I also look into the 64x to update that one as well? / would it be useful?

**Comment by danielga** [comment]
I've rebased the x86-64 branch onto master and got a couple of conflicts with your changes. I've made a backup branch over at [`x86-64-branch-backup`](https://github.com/danielga/sourcesdk-minimal/tree/x86-64-branch-backup).

**Pull request merged**